xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -252,6 +253,54 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 20]> + Into<[u8; 20]>
 	}
 }
 
+/// Constant derivation function for Tinkernet Multisigs.
+/// Uses the Tinkernet genesis hash as a salt.
+pub fn derive_tinkernet_multisig<AccountId: Decode>(
+	id: u128,
+) -> Result<AccountId, <u32 as TryFrom<u128>>::Error> {
+	Ok(AccountId::decode(&mut TrailingZeroInput::new(
+		&(
+			// The constant salt used to derive Tinkernet Multisigs, this is Tinkernet's genesis hash.
+			H256([
+				212, 46, 150, 6, 169, 149, 223, 228, 51, 220, 121, 85, 220, 42, 112, 244, 149, 243,
+				80, 243, 115, 218, 162, 0, 9, 138, 232, 68, 55, 129, 106, 210,
+			]),
+			// The actual multisig integer id.
+			u32::try_from(id)?,
+		)
+			.using_encoded(blake2_256),
+	))
+	.expect("infinite length input; no invalid inputs for type; qed"))
+}
+
+/// Convert a Tinkernet Multisig `MultiLocation` value into a local `AccountId`.
+pub struct TinkernetMultisigAsAccountId<AccountId>(PhantomData<AccountId>);
+impl<AccountId: Decode + Clone> Convert<MultiLocation, AccountId>
+	for TinkernetMultisigAsAccountId<AccountId>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		match location {
+			MultiLocation {
+				parents: _,
+				interior:
+					X3(
+						// Tinkernet ParaId.
+						Parachain(2125),
+						// Pallet INV4, from which the multisigs originate.
+						PalletInstance(71),
```

Comment:
Put these hardcoded literals into constants, so that the comments are unnecessary here.

Reply:
Should I declare the constants above `pub fn derive_tinkernet_multisig` or on the top of the file?

Reply:
I declared them on the top, let me know if you'd rather I change it.